### PR TITLE
marcidy/donate_deploy

### PR DESCRIPTION
**Pull request by marcidy**
This combines the deployment of donate with the initialization of the database.

### CODE REVIEW

**Comment by marcidy** [comment]
@superq Poke

**Comment by SuperQ** [comment]
Looking good so far.

Let's add an attr/env to allow changing the `FLASK_ENV`.

In defaults/main.yml:
```yaml
pydonate_env: development
```

Then in the dot_env template:
```console
FLASK_ENV="{{ pydonate_env | upper }}"
```

**Comment by marcidy** [comment]
Ok, done, deploys fine, ran a test donation

File: roles/pydonate/templates/pydonate.service [hunk 1]
```diff
@@ -2,10 +2,12 @@
 Description=Python donate.noisebridge.net implementation
 
 [Service]
-WorkingDirectory=/srv/{{ pydonate_user }}/code
-ExecStart = "/srv/{{ pydonate_user }}/code/venv/bin/flask" "run"
+WorkingDirectory={{ pydonate_base }}
+ExecStart = "{{ pydonate_base }}/venv/exec" "flask" "run"
```
Comment (SuperQ) [inline roles/pydonate/templates/pydonate.service]: Can you fix the inconsistent use of spaces here?

```ini
ExecStart="{{ pydonate_base }}/venv/exec" "flask" "run"
```

Comment (marcidy) [reply roles/pydonate/templates/pydonate.service]: done

Comment (marcidy) [reply roles/pydonate/templates/pydonate.service]: errr, NOW it's done.  @rizend found an issue with the deployment. I didnt' test it "from scratch" and will do so in the future.

**Comment by SuperQ** [comment]
I'm not sure what the `venv_exec` wrapper is for. It should work to call all the various things in the venv bin directory directly.

**Comment by marcidy** [comment]
I reverted to calling venv/bin/<whatever> directly.  it was a convenience script to activate the venv and run a command.  either way works, and it seems like more people are comfortable with the venv/bin<whatever> method.

I retested deployment and it looks good to me.  other scripts touched to reflect the reversion, no longer need venv/exec, and revert the service definition to use venv/bin/flask.

**Pull request merged**